tutorials/hentropy_search.ipynb

```diff
@@ -0,0 +1,454 @@
FEJyfjpbffRnTLlnj/5ZexesMGdB82DAuWLMErTz6pz+dQA03f9n98meHkC9RYZ6iYoZgYm9PpNKZCmx9TVlaGqKgolN59NyKDgz3eq2rWDMduvhmJcXEIVbjjdvbRMsz9Od8jQDiuSSDmDIxDZodIVWMn4uJFZZk7TByOr3bqvnRJ/S7cVVXyAoOrq92/q6raWhw7fRqJv/2GUL6YnpMnyXbcLigAkpLIrs9YPLR0OV244Plzs2ba9S1nT6eDB8k3pyR1OynZ8kBthpMZ3E1q6s34qZgpq6lB1EcfobS0FJGR2q+rzH3iiSdKERqqrv+qqjIsWBCl21itCo2hsQCZHSIxLKEpcvMrUVRZh5hwl5tJN8sMFyamRq6oCQ+Xvx2AljBuJ7WixmpoIWbYIqZ9+4bvT5zwfE+tuOnYUZ6oKSgw747b/oyfixmKf0AFjVFUVQEBMjx8nOwge4BNu9RsJbDTuWmdFf+FEStsEcNGSNxoabURIinJ/3YdtwJmFjP798tr7+sihhRdoYJGT/bvB1q3bsjRI7UUlJd7ph1rVdJfLYyoocXjzIma7CYhi4wYTDtG2KgRNcXF5FYaK+PrYnpy3U1mFDNcEZOWRnZeTo7nuVTc+B1U0GgNd7L17euqICtHALCFD1vcmEHYUFHjf0hZZaRo316dqGHcTlZAiwrBVrmRmk3MsNdWUhHDhn0OW9xY5e9BkYQKGi05cAAIDFQ22YRgxI2ZhI2VRI3V42hIA4KVpj+rFTMMbGsNYIwLiiKfwYOlrTTsGjNmEDOM8NByXWX6YgsbtdXJKT6HChotSUlxWWP0wGzChslcEour8WVAcGNErrtJKzHDRqm1hjQ4mImjoYHB+mAmq4weQoYLW9gcOKDfdSiGYKk6NCUlJRg3bhwiIyPRrFkzTJkyBeUixdOOHz8Om83G+1q9erW7Hd/7q1atMuIjySciokHcXLrk+xL/jLCprGx4MT+z36d4QpqyTYIS64weYoaB6ZOb7k0xD3v2NIgX5ntmKwNfi5n9+40RM2zS0lwPpH7OokWLkJCQgNDQUKSmpiJXour1hQsXMH36dMTFxSEkJARdunTBBqYSvAmxlIVm3LhxyM/Px6ZNm1BbW4tJkybhvvvuw0pOZVqG+Ph45Ofnexx755138PLLL2PEiBEex9977z1kZma6f25mdpM512JjBmsN0OCK8hch48tNKgsKyNvKsc7oKWYYlFpqtAwOPnlS3y0QrAojWLZtM5dFBjBeyDQiPvnkE2RlZWHJkiVITU3FwoULkZGRgT///BMxMTFe7WtqajBs2DDExMRgzZo1aNu2LU6cOGHqe6NlBM2BAweQnZ2NnTt3on///gCAN998EzfccANeeeUVtLm8HxAbu92OWI5peu3atbj99tsRwYmpaNasmVdbSxARYQ5Rw+AvQsYskBbUI8UIMcMgV9TICQ6WqkfTq5eruJ4UffsqK66nBF8X1ONiFhEDqA/4pUjy2muvYerUqZg0aRIAYMmSJVi/fj2WLVuGJ554wqv9smXLUFJSgu3btyMoKAgAkJCQYOSQZWMZl1NOTg6aNWvmFjMAkJ6ejoCAAOzYsYOoj7y8POzevRtTpkzxem/69Olo1aoVUlJSsGzZMogVUK6urkZZWZnHy6cwbigzuKAkeH/NGjTr3dvXw/Af5KRqGylmGJRcS8qFprXIo/gWtlWGihlZcO9D1dXVvO1qamqQl5eH9PR097GAgACkp6cjR6CUwJdffom0tDRMnz4drVu3Rs+ePTFv3jw4HA5dPosWWEbQFBQUeJnFAgMD0aJFCxQQmubfffdddOvWDQMGDPA4/uyzz+LTTz/Fpk2bcMstt+CBBx7AmyIbJc6fPx9RUVHuV3x8vPwPpAfs2BoNydm1C/aOHTFy8mRZ5yUMGoSFy5Z5HLvjxhtxaMsWLYdHIcEXYoahfXvyeJrGvEt4Y6QRuphKS13TQc2rtNTVV3x8vMe9aP78+bzXPHv2LBwOB1q3bu1xvHXr1oL3z6NHj2LNmjVwOBzYsGEDnn76abz66qt4/vnnVX1+PfG5y+mJJ57Aiy++KNrmgAbR55cuXcLKlSvx9NNPe73HPtavXz9UVFTg5Zdfxj//+U/evmbNmoWsrCz3z2VlZfqLmnoHUJgHVJ4FwlsBrZOBAJ69oXRwQb37ySd4cMIEvPvppzhTWIg2nEkhh7DQUISFhmoyLvKLmix1myQgmCRdW24hPV+IGfa1T5wgj6chiaWh2yBYm0YoZrTm1KlTHns5hWiYZVtfX4+YmBi88847sNvtSE5OxunTp/Hyyy9jzpw5ml1HS3xuoXnkkUdw4MAB0VeHDh0QGxuLoqIij3Pr6upQUlJCFPuyZs0aVFZWYvz48ZJtU1NT8ffffwua70JCQhAZGenxkqS62rX9AftFyvFNwKfDgQ2Tga2Pub5+Otx1nA8NLTXlFRX4ZP16TBs3DiOvuw7vr1nj8f5X332Hq0aPRmjXrmh15ZW4+R//AAAMGTsWJ06fxsPPPQdbYiJsiYkA+F1Oiz/6CB2vvRbBXbqg6/XX48PPP/d435aYiP+tWoWb//EPhHfrhs7XXYcvNwl8dq2RuzmlGblwwbdihg2JpYZEpJG6nU6eJGtHMQ4mi0lvF9OhQ54vP4R7HxISNK1atYLdbkdhYaHH8cLCQsH7Z1xcHLp06QI7a1Plbt26oaCgADVM9XuT4XNBEx0djaSkJNFXcHAw0tLScOHCBeTl5bnP3bJlC+rr65Gamip5nXfffRc33XQTogkyKHbv3o3mzZsrV7vsSXT8OFB3eZfsJk08X1yBwydyjm8CNmcBFZ7/iKgoch3XWdR8un49kjp2RNeOHXH3mDFYtnq1O75o/ZYtuPn++3HDkCH4bf16bP7oI6RcLtj1+ZIluCIuDs8+/DDyc3ORL5AeuHbjRsx89lk8cu+9+D07G/+46y5MeuwxfM/x68594w3cPnIk9n7zDW4YMgTjHn4YJY05LZ
```

Comment:
It's not exactly clear in the output what the iterations are. Could you explain / delineate those better in the outputs?

Reply:
Similar to the two questions above, I wasn't able to see what lines in the code this question is about. Could you please help me to clarify this?

Reply:
Oh I just mean that you're running "iterations" here in the tutorial but it's not exactly clear what these are and why they're being run. Can you preface the actual run with some more comments that explain what is happening? Just basically some more guiding the reader so they know what's going on at a higher level.